### ISSUE TITLE
Errors in Etherpad Docker, broken Shared Notes 

### ISSUE THREAD

**Issue by mghadam** [alangecker/bigbluebutton-docker#76]
I can't find any icons for using shared notes / etherpad in the conference room. This is while shared notes are enabled in ./mod/html5/settings.yml file in the default settings and etherpad docker is also running, I can also browse /pad in my web browser however there is absolutely nowhere to activate or use shared notes.

This is a screenshot from bigbluebutton-docker: 
![image](https://user-images.githubusercontent.com/10244293/98328513-2a386480-200b-11eb-8c45-4b0f8457d3b7.png)

And this is BigBlueButton on Ubuntu without docker: 
![image](https://user-images.githubusercontent.com/10244293/98328562-40debb80-200b-11eb-9718-bd723b598dbf.png)

I wonder why no Shared Notes icon is shown in bigbluebutton-docker while it is already enabled in all config files by default.

**update:**
I found the following errors in "docker-compose logs  -f | grep -i error  | grep -i "note\|etherpad""

```
etherpad_1     | [2020-11-05 19:53:18.949] [ERROR] console - Failed to load '/opt/etherpad-lite/node_modules/ep_bigbluebutton_patches/index:init' for 'ep_bigbluebutton_patches/bbb_hooks/hooks/init_ep_bigbluebutton_patches': Not a function
etherpad_1     | [2020-11-05 19:53:18.976] [ERROR] console - Failed to load 'ep_bigbluebutton_patches/index:init' for 'ep_bigbluebutton_patches/bbb_hooks/hooks/init_ep_bigbluebutton_patches': Not a function
html5_1        | error: Could not create note for 2e7ab4c844291de8b2ee4a6f2fc4c1b462129613-1604606295658: Error: Request failed with status code 503

```

and here is the full log for etherpad docker:

```
etherpad_1     | All relative paths will be interpreted relative to the identified Etherpad base dir: /opt/etherpad-lite
etherpad_1     | Random string used for versioning assets: 327c303a
etherpad_1     | [2020-11-05 19:53:17.121] [DEBUG] AbsolutePaths - Relative path "settings.json" can be rewritten to "/opt/etherpad-lite/settings.json"
etherpad_1     | [2020-11-05 19:53:17.122] [DEBUG] AbsolutePaths - Relative path "credentials.json" can be rewritten to "/opt/etherpad-lite/credentials.json"
etherpad_1     | settings loaded from: /opt/etherpad-lite/settings.json
etherpad_1     | No credentials file found in /opt/etherpad-lite/credentials.json. Ignoring.
etherpad_1     | Unknown Setting: 'cookie'. This setting doesn't exist or it was removed
etherpad_1     | [2020-11-05 19:53:17.134] [INFO] console - Using skin "bigbluebutton" in dir: /opt/etherpad-lite/src/static/skins/bigbluebutton
etherpad_1     | [2020-11-05 19:53:17.134] [INFO] console - Session key file "/opt/etherpad-lite/SESSIONKEY.txt" not found. Creating with random contents.
etherpad_1     | [2020-11-05 19:53:18.765] [INFO] APIHandler - Api key file read from: "/tmp/apikey"
etherpad_1     | [2020-11-05 19:53:18.949] [ERROR] console - Failed to load '/opt/etherpad-lite/node_modules/ep_bigbluebutton_patches/index:init' for 'ep_bigbluebutton_patches/bbb_hooks/hooks/init_ep_bigbluebutton_patches': Not a function
etherpad_1     | [2020-11-05 19:53:18.972] [INFO] console - Installed plugins: ep_bigbluebutton_patches@0.0.1, ep_delete_after_delay_lite@0.0.5, ep_redis_publisher@0.0.1
etherpad_1     | [2020-11-05 19:53:18.976] [ERROR] console - Failed to load 'ep_bigbluebutton_patches/index:init' for 'ep_bigbluebutton_patches/bbb_hooks/hooks/init_ep_bigbluebutton_patches': Not a function
etherpad_1     | [2020-11-05 19:53:18.978] [INFO] console - Report bugs at https://github.com/ether/etherpad-lite/issues
etherpad_1     | [2020-11-05 19:53:18.979] [INFO] console - Your Etherpad version is 1.8.6 (299bd96)
etherpad_1     | [2020-11-05 19:53:19.247] [INFO] console - You can access your Etherpad instance at http://0.0.0.0:9001/
etherpad_1     | [2020-11-05 19:53:19.247] [WARN] console - Admin username and password not set in settings.json.  To access admin please uncomment and edit 'users' in settings.json
etherpad_1     | Node worker_threads not available. Trying to fall back to tiny-worker polyfill...
etherpad_1     | Node worker_threads not available. Trying to fall back to tiny-worker polyfill...
etherpad_1     | [2020-11-05 20:53:18.949] [INFO] console - New loop
etherpad_1     | [2020-11-05 21:53:18.951] [INFO] console - New loop
```

**Comment by mghadam** [comment]
This is a screenshot from bigbluebutton-docker: 
![image](https://user-images.githubusercontent.com/10244293/98328513-2a386480-200b-11eb-8c45-4b0f8457d3b7.png)

And this is BigBlueButton on Ubuntu without docker: 
![image](https://user-images.githubusercontent.com/10244293/98328562-40debb80-200b-11eb-9718-bd723b598dbf.png)

I wonder why no Shared Notes icon is shown in bigbluebutton-docker while it is already enabled in all config files by default.

**Comment by alangecker** [comment]
I currently can't reproduce this problem, but I also get this error messages:
```
etherpad_1             | [2020-11-06 07:23:04.882] [ERROR] console - Failed to load '/opt/etherpad-lite/node_modules/ep_bigbluebutton_patches/index:init' for 'ep_bigbluebutton_patches/bbb_hooks/hooks/init_ep_bigbluebutton_patches': Not a function
etherpad_1             | [2020-11-06 07:23:04.908] [ERROR] console - Failed to load 'ep_bigbluebutton_patches/index:init' for 'ep_bigbluebutton_patches/bbb_hooks/hooks/init_ep_bigbluebutton_patches': Not a function
```
I forget there to clean up the reference after removing a hook - should not affect the functionality of etherpad at all.


but this line in your logs is interesting indeed:
```
html5_1        | error: Could not create note for 2e7ab4c844291de8b2ee4a6f2fc4c1b462129613-1604606295658: Error: Request failed with status code 503
```
could you provide us with more context? there are probably more helpful lines in the logs at that time

this command could help: `./scripts/compose logs -f | grep -A 5 -B 20 "Could not create note"` (additionally prints out the previous 20 and following 5 lines)

**Comment by mghadam** [comment]
I found the problem, docker was trying to connect to etherpad docker through the proxy I had set for docker images. Fixed by adding `postgres, etherpad, 0.0.0.0` to noProxy section of my docker config file:

`     "noProxy": "localhost, 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16, 127.0.0.1, postgres, etherpad, 0.0.0.0"
`
